fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java

```diff
@@ -354,13 +355,32 @@ public List<Rule> buildRules() {
             RuleType.BINDING_SORT_SLOT.build(
                 logicalSort(logicalProject()).when(Plan::canBind).thenApply(ctx -> {
                     LogicalSort<LogicalProject<GroupPlan>> sort = ctx.root;
+                    Set<Slot> orderSlots = Sets.newHashSet();
```

Comment:
should we also update BINDING_SORT_SLOT rule on pattern `logicalSort(logicalHaving(logicalProject()))`

Reply:
I think its ok on pattern sort(having(project)).
This pr makes binding more strict, and it is applicable to all patterns.